trace/trace.go

```diff
@@ -0,0 +1,68 @@
+package trace
+
+import (
+	"time"
+)
+
+////////////////////////////////////////////////////////////////////////////////
+
+type ClusterTrace struct {
+	// TopoChanged is called when the cluster's topology changed.
+	TopoChanged func(ClusterTopoChanged)
+	// GotResponse is called after the radix.Do executed.
+	GotResponse func(ClusterGotResponse)
+}
+
+type ClusterNodeInfo struct {
+	Addr      string
+	Slots     [][2]uint16
+	IsPrimary bool
+}
+
+type ClusterTopoChanged struct {
+	Added   []ClusterNodeInfo
+	Removed []ClusterNodeInfo
+}
+
+type ClusterGotResponse struct {
```

Comment:
I don't really like this one. Instead of tracing just how many tries a command needed, we could just trace `ASK` and `MOVED` directly. What do you think?

Reply:
What I was trying to get is just for `MOVED` command (which is a great monitoring metric that can be used for checking cluster topology has been changed). I can change this to handle `MOVED` command only If you think `ClusterGotResponse` is too broad.

Reply:
I think a more usable version of this might look like:

```go
type ClusterRedirect struct {
    Addr string
    Key string // if possible
    Moved, Ask bool
    RedirectCount int // basically RetryCount
}
```

I don't think it's necessary to be tracing on _every_ response, only the redirect ones.

Reply:
So basically, nearly the same as you have now, but moving it to (here)[https://github.com/mediocregopher/radix/pull/100/files#diff-5012ac478fddb03fa409c41a16a72b0dL424}

Reply:
Agree. Tracing the redirect response makes sense to me as well. Thanks. I'll fix it.

Reply:
For the `RedirectCount`, It seems that radix library only handles **remain** retry counts for the response. Should I change it to `RemainRetryCount` or fix the codes to keep actual retried count?

Reply:
I think it'd be fine to pass `doAttempts - attempts` as the `RedirectCount` value.

Reply:
Okay, I'll fix it.